irods/test/data_obj_test.py

```diff
@@ -1706,6 +1708,35 @@ def test_set_and_access_data_comments__issue_450(self):
             finally:
                 d.unlink(force = True)
 
+    def _auto_close_test(self, data_object_path, content):
+        d = None
+        try:
+            d = self.sess.data_objects.get(data_object_path)
+            self.assertEqual(int(d.replicas[0].status), 1)
+            self.assertEqual(d.open('r').read().decode(), content)
+        finally:
+            if d: d.unlink(force = True)
+
+    def test_data_objects_auto_close_on_process_exit__issue_456(self):
+        program = os.path.join(test_modules.__path__[0], 'test_auto_close_of_data_objects__issue_456.py')
+        # Use the currently running Python interpreter binary to run the script in the child process.
+        p = subprocess.Popen([sys.executable,program], stdout=subprocess.PIPE)
+        data_object_path, expected_content = p.communicate()[0].decode().split()
+        self._auto_close_test(data_object_path, expected_content)
+
+    def test_data_objects_auto_close_on_function_exit__issue_456(self):
+        import irods.test.modules.test_auto_close_of_data_objects__issue_456 as test_module
+        data_object_path, expected_content = test_module.test(return_locals = ('name','expected_content'))
+        self._auto_close_test(data_object_path, expected_content)
+
+    @unittest.skipIf(helpers.configuration_file_exists(),"test would overwrite pre-existing configuration.")
+    def test_settings_save_and_autoload__issue_471(self):
+        import irods.test.modules.test_saving_and_loading_of_settings__issue_471 as test_module
+        truth = int(time.time())
+        test_output = test_module.test(truth)
```

Comment:
What's the motivation behind using an integer?
Are we relying on the positive integer being converted to true?

Reply:
Yes, it will evaluate True; and by using an integer I'm guaranteeing a unique value will be rendered into the config file.

Reply:
Ah, got it.